modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceAddressLocalServiceImpl.java

```diff
@@ -579,6 +649,65 @@ public int compare(Address address1, Address address2) {
 		};
 	}
 
+	private GroupByStep _getGroupByStep(
```

Comment:
I don't see any filter on the address type.
If we are searching for billing addresses, it's fine that we check for the country to be enabled as billing, but also the address itself must be a billing-address.
A shipping only address that is linked to a billing alowed country will be retrieved by this method, but it should not.

Reply:
In older method we have filters like
"listTypeIds",
					new long[] {
						CommerceAddressImpl.toAddressTypeId(
							CommerceAddressConstants.ADDRESS_TYPE_BILLING),
						CommerceAddressImpl.toAddressTypeId(
							CommerceAddressConstants.
								ADDRESS_TYPE_BILLING_AND_SHIPPING)
					}

Reply:
@luca-pellizzon , just to confirm, if (Billing), we want both ADDRESS_TYPE_BILLING **AND** ADDRESS_TYPE_BILLING_AND_SHIPPING to show up right? Same for if (Shipping)

Reply:
See commit: https://github.com/liferay-commerce/liferay-portal/pull/3448/commits/75874b1a13a8f646199910022c9e0e3fd58fad24

Reply:
Yes, addresses that are ADDRESS_TYPE_BILLING_AND_SHIPPING should be considered for both methods, the code looks fine now in that sense.